biogeochem/EDPhysiologyMod.F90

```diff
@@ -408,135 +408,185 @@ subroutine phenology( currentSite, bc_in )
+    currentSite%vegtemp_memory(2:num_vegtemp_mem) = currentSite%vegtemp_memory(1:num_vegtemp_mem-1)
+    currentSite%vegtemp_memory(1) = temp_in_C
+
     !count number of days for leaves off
     ncolddays = 0
-    do i = 1,senes
-       if (currentSite%last_n_days(i) < ED_val_phen_coldtemp)then
+    do i_tmem = 1,num_vegtemp_mem
+       if (currentSite%vegtemp_memory(i_tmem) < ED_val_phen_coldtemp)then
           ncolddays = ncolddays + 1
        endif
     enddo
 
     ! Here is where we do the GDD accumulation calculation
     !
     ! reset GDD on set dates
-    if (t == gddstart)then
-       currentSite%ED_GDD_site = 0._r8
+    if (hlm_day_of_year == gddstart)then
+       currentSite%grow_deg_days = 0._r8
     endif
     !
     ! accumulate the GDD using daily mean temperatures
     if (bc_in%t_veg24_si .gt. tfrz) then
-       currentSite%ED_GDD_site = currentSite%ED_GDD_site + bc_in%t_veg24_si - tfrz
+       currentSite%grow_deg_days = currentSite%grow_deg_days + bc_in%t_veg24_si - tfrz
     endif
     
+    ! Calculate the number of days since the leaves last came on 
+    ! and off. If this is the beginning of the simulation, that day might
+    ! not had occured yet, so set it to last year to get things rolling
+
+    if (model_day_int < currentSite%cleafoffdate) then
+       dayssincecleafoff = model_day_int - (currentSite%cleafoffdate - 365)
+    else
+       dayssincecleafoff = model_day_int - currentSite%cleafoffdate
+    end if
+
+    if (model_day_int < currentSite%cleafondate) then
+       dayssincecleafon = model_day_int - (currentSite%cleafondate-365)
+    else
+       dayssincecleafon = model_day_int - currentSite%cleafondate
+    end if
+
+
 
-    timesinceleafoff = hlm_model_day - currentSite%leafoffdate
     !LEAF ON: COLD DECIDUOUS. Needs to
     !1) have exceeded the growing degree day threshold 
     !2) The leaves should not be on already
-    !3) There should have been at least on chilling day in the counting period.  
-    if (currentSite%ED_GDD_site > gdd_threshold)then
-       if (currentSite%is_cold) then
-          if (currentSite%ncd >= 1) then
-             ! NOTE(bja, 2015-01) should leafondate = model_day to be consistent with leaf off?
-             currentSite%is_cold = .false.      
-             currentSite%leafondate = t         ! record leaf on date   
-             if ( debug ) write(fates_log(),*) 'leaves on'
-          endif !ncd
-       endif !status
+    !3) There should have been at least one chilling day in the counting period.  
+    !   this prevents tropical or warm climate plants that are "cold-deciduous"
+    !   from ever re-flushing after they have reached their maximum age (thus
+    !   preventing them from competing
+
+    if ( (currentSite%cstatus == 1 .or. currentSite%cstatus == 0) .and. &
+         (currentSite%grow_deg_days > gdd_threshold) .and. &
+         (currentSite%nchilldays >= 1)                    .and. &
+         (dayssincecleafoff > ED_val_phen_mindayson))  then
```

Comment:
@rgknox, why does the logic need to check the number of days leaves have been off against the minimum days on parameter?

Reply:
This is to prevent fickle plants from resprouting leaves if the local climate is in the process of getting cold, but has a spurious warm day.  However, since we do zero the growing degree days when leaves are dropped, this really doesn't seem necessary...